#!/usr/bin/env python
PACKAGE = "visual_servoing_davis"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("eHarris_threshold", double_t, 0, " The inverse ratio of resolution",    4, 0,   10)
gen.add("eHarris_window_size", int_t, 0, " The inverse ratio of resolution",    4, 0,   25)
gen.add("eHarris_queue_size", int_t, 0, " The inverse ratio of resolution",    25, 0,   50)
gen.add("Harris_threshold", int_t, 0, " The inverse ratio of resolution",    10, 0,   255)
gen.add("Harris_block_size", int_t, 0, " The inverse ratio of resolution",    5, 0,   10)
gen.add("Harris_aperture_size", int_t, 0, " The inverse ratio of resolution",    7, 0,   10)
gen.add("target_velocity", double_t, 0, " The inverse ratio of resolution",    0.35, 0,   2.0)
gen.add("center_offset_threshold", int_t, 0, " The inverse ratio of resolution",   3, 0,   10)
gen.add("heatmap_time_decay_factor", double_t, 0, " The inverse ratio of resolution",    0.6, 0,   1)
gen.add("heatmap_corner_weight", double_t, 0, " The inverse ratio of resolution",    0.15, 0,   1)
gen.add("heatmap_threshold", double_t, 0, " The inverse ratio of resolution",    0.65, 0,   1)
gen.add("heatmap_dilate_kernel_size", int_t, 0, " The inverse ratio of resolution",    25, 0,   35)
gen.add("tracking_corner_var", double_t, 0, " The inverse ratio of resolution",    800, 0,   1000)
gen.add("tracking_event_var", double_t, 0, " The inverse ratio of resolution",    1500, 0,   3000)
gen.add("tracking_likelihood_thresh", double_t, 0, " The inverse ratio of resolution",    0.4, 0,   1)
gen.add("tracking_kp", double_t, 0, " The inverse ratio of resolution",    0.07, 0,   0.1)
gen.add("tracking_kd", double_t, 0, " The inverse ratio of resolution",    0.001, 0,   1)
gen.add("false_tracking_thresh", int_t, 0, " The inverse ratio of resolution",    3, 0,   15)
gen.add("tracking_err_thresh", int_t, 0, " The inverse ratio of resolution",    30, 0,   50)

exit(gen.generate(PACKAGE, "visual_servoing_davis", "VSCfg"))
